File: app/src/main/java/uk/co/alt236/apkcompare/app/comparators/ApkComparator.java
package uk.co.alt236.apkcompare.app.comparators;

import uk.co.alt236.apk.Apk;
import uk.co.alt236.apkcompare.app.comparators.results.ComparisonResult;

import java.util.List;


public interface ApkComparator {
    List<ComparisonResult> compare(Apk file1, Apk file2);
}

File: app/src/main/java/uk/co/alt236/apkcompare/app/comparators/ApkComparator.kt
package uk.co.alt236.apkcompare.app.comparators

import uk.co.alt236.apk.Apk
import uk.co.alt236.apkcompare.app.comparators.results.ComparisonResult

interface ApkComparator {
    val name: String
    fun compare(apk1: Apk, apk2: Apk): List<ComparisonResult>
}